src/components/timed-button/timed-button.tsx

```diff
@@ -39,19 +42,33 @@ export const TimedButton: FC<TimedButtonProps> = ({
     }
   };
 
-  const buttonText = status === "active" ? props.activeLabel : props.label;
-
   return (
     <button
       onClick={handleClick}
       className={twMerge(
-        'focus-ring-neutral inline-flex items-center px-4 py-2 cursor-pointer rounded-2xl hover:bg-neutral-100 active:bg-neutral-100"',
+        "focus-ring-neutral inline-flex items-center px-4 py-2 cursor-pointer rounded-2xl hover:bg-neutral-100 active:bg-neutral-100 transition-colors duration-[350ms] ease-in-out",
         status === "processing" && "cursor-wait bg-neutral-100",
         status === "active" && "cursor-default bg-neutral-100",
       )}
     >
-      {props.icon && <span className="mr-2">{props.icon}</span>}
-      {buttonText}
+      {icon && <span className="mr-2">{icon}</span>}
+      <span className="relative flex justify-center items-center">
+        {status !== "active" && (
+          <span className="transition-opacity duration-[350ms] ease-in-out block text-center whitespace-nowrap opacity-100">
+            {label}
+          </span>
+        )}
+        <span
+          className={twMerge(
+            "transition-opacity duration-[350ms] ease-in-out block text-center whitespace-nowrap",
```

Comment:
it seems like we always use duration-[350ms].
should we make a class for this instead of always use an arbitrary value?